internal/awsservice/cloudwatchlogs.go

```diff
@@ -134,30 +72,50 @@ func DeleteLogGroup(logGroupName string) {
 	}
 }
 
-// ValidateLogsInOrder takes a log group, log stream, a list of specific log lines and a timestamp.
-// It should query the given log stream for log events, and then confirm that the log lines that are
-// returned match the expected log lines. This also sanitizes the log lines from both the output and
-// the expected lines input to ensure that they don't diverge in JSON representation (" vs ')
-func ValidateLogsInOrder(t *testing.T, logGroup, logStream string, logLines []string, since time.Time) {
-	log.Printf("Checking %s/%s since %s for %d expected logs", logGroup, logStream, since.UTC().Format(time.RFC3339), len(logLines))
-	cwlClient, clientContext, err := getCloudWatchLogsClient()
+// ValidateLogs queries a given LogGroup/LogStream combination given the start and end times, and executes an
+// arbitrary validator function on the found logs.
+func ValidateLogs(logGroup, logStream string, since, until *time.Time, validator func(logs []string) bool) (bool, error) {
+	log.Printf("Checking %s/%s\n", logGroup, logStream)
+
+	foundLogs, err := getLogsSince(logGroup, logStream, since, until)
 	if err != nil {
-		t.Fatalf("Error occurred while creating CloudWatch Logs SDK client: %v", err.Error())
+		return false, err
 	}
 
-	sinceMs := since.UnixNano() / 1e6 // convert to millisecond timestamp
+	return validator(foundLogs), nil
+}
+
+// getLogsSince makes GetLogEvents API calls, paginates through the results for the given time frame, and returns
+// the raw log strings
+func getLogsSince(logGroup, logStream string, since, until *time.Time) ([]string, error) {
+	foundLogs := make([]string, 0)
+
+	cwlClient, clientContext, err := getCloudWatchLogsClient()
+	if err != nil {
+		return foundLogs, err
+	}
 
 	// https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_GetLogEvents.html
 	// GetLogEvents can return an empty result while still having more log events on a subsequent page,
 	// so rather than expecting all the events to show up in one GetLogEvents API call, we need to paginate.
 	params := &cloudwatchlogs.GetLogEventsInput{
 		LogGroupName:  aws.String(logGroup),
 		LogStreamName: aws.String(logStream),
-		StartTime:     aws.Int64(sinceMs),
 		StartFromHead: aws.Bool(true), // read from the beginning
 	}
 
-	foundLogs := make([]string, 0)
+	var sinceMs int64
+	if since != nil {
+		sinceMs = since.UnixNano() / 1e6 // convert to millisecond timestamp
+		params.StartTime = aws.Int64(sinceMs)
```

Comment:
Would be better to pass startTime and EndTime as requirement instead of using `sinceMs` and `untilMs` since there will be rarely a case there are logs within the same time. (force better than lenient here)

Reply:
What does this mean? What do you mean rarely a case?

Reply:
There are scenarios where we don't know the start time or the end time. So no.